content/blog/2018/06/2018-06-26-new-api-token-system.adoc

```diff
@@ -0,0 +1,98 @@
+
+image::/images/post-images/2018-06-26-new-api-token-system/token_usage.png[title="Token usage tracking", role="text-center", width=800, link="/images/post-images/2018-06-26-new-api-token-system/token_usage.png"]
+
+* You can revoke API tokens.
+When you know that you are not using a given token anymore, you can revoke it to reduce the risk of it getting used by unauthorized users.
+Since you can have multiple API tokens, this allows fine-grained control over which scripts, hosts, or applications are allowed to use Jenkins as a given user.
+
+### Migrating to new API tokens
+
+To help administrators migrate their instances progressively, the legacy behavior is still available, while new system is also usable.
+
+On the user configuration page, the legacy token is highlighted with a warning sign,
+explaining that users should revoke it and generate a new one (if needed) to increase security.
+
+image::/images/post-images/2018-06-26-new-api-token-system/legacy_renewal.gif[title="Legacy token renewal still possible", role="text-center", width=800, link="/images/post-images/2018-06-26-new-api-token-system/legacy_renewal.gif"]
+
+#### New options for administrators
+
+In order to let administrators control the pace of migration to the new API token system,
+we added two global configuration options in the "Configure Global Security" page in the brand new "API Token" section:
+
+* An option to disable the creation of legacy API tokens on user creation.
+
+* An option to disable the recreation of legacy API tokens by users, forcing them to only use the new, unrecoverable API tokens.
+
+Both options are disabled by default for new installations (the safe default), while they're enabled when Jenkins is upgraded from before 2.129.
+
+image::/images/post-images/2018-06-26-new-api-token-system/security_configuration_options.png[title="Security Configuration options", role="text-center", link="/images/post-images/2018-06-26-new-api-token-system/security_configuration_options.png"]
+
+image::/images/post-images/2018-06-26-new-api-token-system/legacy_removal.gif[title="Remove legacy token and disable the re-creation", role="text-center", width=800, link="/images/post-images/2018-06-26-new-api-token-system/legacy_removal.gif"]
+
+#### New administrator warnings
+
+When upgrading to Jenkins 2.129, an administrative monitor informs admins about the new options described above, and recommend disabling them.
+
+Another administrative warnings shows up if at least one user still has a legacy API token.
+It provides central control over legacy tokens still configured in the Jenkins instance, and allows revoking them all.
+
+image::/images/post-images/2018-06-26-new-api-token-system/monitor_screen.png[title="Legacy token monitoring page", role="text-center", width=800, link="/images/post-images/2018-06-26-new-api-token-system/monitor_screen.png"]
+
+### Summary
+
+Jenkins API tokens are now much more flexible: They allow and even encourage better security practices.
```

Comment:
"The new ..." ?

Reply:
I mean the system as a whole, not a specific version of it. Hence the "now". Could go either way, I prefer this one.